File: backend/app/schemas/saved_prompt.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class SavedPromptCreate(BaseModel):
    content: str = Field(min_length=1, max_length=2000)


class SavedPromptUpsert(BaseModel):
    content: str = Field(min_length=1, max_length=2000)


class SavedPromptResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    slot: int
    content: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/savedPrompt.ts
export interface SavedPrompt {
  id: string;
  slot: number;
  content: string;
  created_at: string;
  updated_at: string;
}

export interface SavedPromptCreate {
  content: string;
}

export interface SavedPromptUpsert {
  content: string;
}
